pkg/platform/kvmlibvirt/template.go

```diff
@@ -0,0 +1,186 @@
+  {{- end }}
+
+  {{- if .Config.ControllerVirtualMemory}}
+  virtual_memory {{ .Config.ControllerVirtualMemory}}
+  {{- end }}
+
+  {{- if .Config.NetworkMTU }}
+  network_mtu = {{.Config.NetworkMTU}}
+  {{- end }}
+
+  {{- if .Config.NetworkIpAutodetectionMethod }}
+  network_ip_autodetection_method = {{ .Config.NetworkIpAutodetectionMethod }}
+  {{- end }}
+
+  {{- if .Config.PodCidr }}
+  pod_cidr = {{.Config.PodCidr }}
+  {{- end }}
+  {{- if .Config.ServiceCidr }}
+  service_cidr = {{.Config.ServiceCidr }}
+  {{- end }}
+
+  {{- if .Config.ClusterDomainSuffix }}
+  cluster_domain_suffix = {{.Config.ClusterDomainSuffix }}
+  {{- end }}
+
+  enable_reporting   = {{.Config.EnableReporting}}
+  enable_aggregation = {{.Config.EnableAggregation}}
+
+  {{- if .Config.ControllerCLCSnippets }}
+  controller_clc_snippets = {{ StringsJoin .Config.ControllerCLCSnippets "\", \"" | printf "[%q]" }}
+  {{- end }}
+
+  disable_self_hosted_kubelet = {{ .Config.DisableSelfHostedKubelet }}
+
+  {{- if .Config.KubeAPIServerExtraFlags }}
+  kube_apiserver_extra_flags = {{ StringsJoin .Config.KubeAPIServerExtraFlags "\", \"" | printf "[%q]" }}
+  {{- end }}
+
+  {{- if .Config.CertsValidityPeriodHours }}
+  certs_validity_period_hours = {{.Config.CertsValidityPeriodHours}}
+  {{- end }}
+}
+
+{{ range $index, $pool := .Config.WorkerPools }}
+module "worker-pool-{{ $index }}" {
+  source = "../terraform-modules/kvm-libvirt/flatcar-linux/kubernetes/workers"
+
+  ssh_keys = {{ StringsJoin $.Config.SSHPubKeys "\", \"" | printf "[%q]" }}
+  machine_domain        = "{{$.Config.MachineDomain}}"
+  cluster_name          = "{{$.Config.ClusterName}}"
+  {{- if $.Config.ClusterDomainSuffix }}
+  cluster_domain_suffix = {{$.Config.ClusterDomainSuffix }}
+  {{- end }}
+  {{- if $.Config.ServiceCidr }}
+  service_cidr = {{$.Config.ServiceCidr }}
+  {{- end }}
+
+  libvirtpool           = module.kvm-libvirt-{{$.Config.ClusterName}}.libvirtpool
+  libvirtbaseid         = module.kvm-libvirt-{{$.Config.ClusterName}}.libvirtbaseid
+  kubeconfig            = module.kvm-libvirt-{{$.Config.ClusterName}}.kubeconfig
+
+  pool_name             = "{{ $pool.Name }}"
+  worker_count          = "{{ $pool.Count}}"
+
+  {{- if $pool.VirtualCPUs }}
+  virtual_cpus = {{ $pool.VirtualCPUs }}
+  {{- end }}
+
+  {{- if $pool.VirtualMemory }}
+  virtual_memory {{ $pool.VirtualMemory }}
+  {{- end }}
+
+  {{- if $pool.Labels }}
+  labels = {{ $pool.Labels }}
+  {{- end }}
+
+  {{- if $pool.CLCSnippets }}
+  clc_snippets = {{ StringsJoin $pool.CLCSnippets "\", \"" | printf "[%q]" }}
+  {{- end }}
+}
+{{- end }}
+
+provider "libvirt" {
+  uri     = "qemu:///system"
+  version = "~> 0.6.0"
+}
+
+provider "ct" {
+  version = "~> 0.5.0"
+}
+
+provider "local" {
+  version = "~> 1.2"
+}
+
+provider "null" {
+  version = "~> 2.1"
+}
+
+provider "template" {
+  version = "~> 2.1"
+}
+
+provider "tls" {
+  version = "~> 2.0"
+}
+
+provider "random" {
+  version = "~> 2.2"
+}
```

Comment:
I guess this can be removed from here and moved to `assets/terraform-modules/kvm-libvirt/flatcar-linux/kubernetes/versions.tf`